compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/Util/IR/UtilDialect.h"
+#include "iree/compiler/Dialect/Util/IR/UtilOps.h"
+#include "iree/compiler/Dialect/Util/IR/UtilTypes.h"
+#include "iree/compiler/InputConversion/Common/PassDetail.h"
+#include "iree/compiler/InputConversion/Common/Passes.h"
+#include "mlir/Dialect/MLProgram/IR/MLProgram.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+struct ImportMLProgramPass : public ImportMLProgramBase<ImportMLProgramPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<IREE::Util::UtilDialect>();
+  }
+  void runOnOperation() override;
+};
+
+class IREETypeConverter : public TypeConverter {
+ public:
+  IREETypeConverter();
+};
+
+// Generic 1:1 conversion pattern which effectively just renames an op.
+// It does not support regions or ops with successors.
+class OneToOneConversionPattern : public ConversionPattern {
+ public:
+  OneToOneConversionPattern(TypeConverter &converter, StringRef srcName,
+                            StringRef targetName, MLIRContext *context,
+                            PatternBenefit benefit)
+      : ConversionPattern(converter, srcName, benefit, context),
+        targetName(targetName) {}
+  LogicalResult matchAndRewrite(
+      Operation *srcOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Type> resultTypes;
+    if (failed(typeConverter->convertTypes(srcOp->getResultTypes(),
+                                           resultTypes))) {
+      return srcOp->emitError()
+             << "could not convert result types to IREE internal types";
```

Comment:
In failure case would op be printed here or should we print out all the types here? (looking at implementation of convertTypes the failed type would be `srcOp->getResultTypes()[resultTypes.size()]` but that's not documented as something we can rely on, I'll ask]